ivy/functional/backends/paddle/experimental/set.py

```diff
@@ -0,0 +1,57 @@
+# global
+from typing import Tuple
+import paddle
+from ivy.func_wrapper import with_unsupported_device_and_dtypes
+
+# local
+import ivy.functional.backends.paddle as paddle_backend
+from . import backend_version
+
+
+@with_unsupported_device_and_dtypes(
+    {"2.4.2 and below": {"cpu": ("complex", "int8", "uint8", "int16", "float16")}},
+    backend_version,
+)
+def intersection(
+    x1: paddle.Tensor,
+    x2: paddle.Tensor,
+    /,
+    *,
+    assume_unique: bool = False,
+    return_indices: bool = False,
+) -> Tuple[paddle.Tensor, paddle.Tensor, paddle.Tensor]:
+    x1 = paddle.reshape(x1, [-1])
+    x2 = paddle.reshape(x2, [-1])
+    if not assume_unique:
+        x1, ind1, _, _ = paddle_backend.unique_all(x1)
+        x2, ind2, _, _ = paddle_backend.unique_all(x2)
+
+    aux = paddle.concat([x1, x2], 0)
+    if return_indices:
+        values_ = paddle.moveaxis(aux, 0, 0)
+        values_ = paddle.reshape(values_, (values_.shape[0], -1))
+        aux_sort_indices = paddle.to_tensor(
+            [i[0] for i in sorted(list(enumerate(values_)), key=lambda x: tuple(x[1]))]
+        )
```

Comment:
please use tensor operations to perform this operation, is there a reason for not doing so?

Reply:
See comment about `paddle.argsort`.